src/main/java/com/code_intelligence/jazzer/mutation/mutator/libfuzzer/LibFuzzerMutatorFactory.java

```diff
@@ -71,7 +71,7 @@ public byte[] read(DataInputStream in) throws IOException {
 
     @Override
     public byte[] readExclusive(InputStream in) throws IOException {
-      return readAllBytes(in);
+      return readNBytes(in, maxLength);
```

Comment:
The current contract of `readExclusive` states that it consumes the remaining input stream which would not be the case if we read only `maxLength` bytes. Maybe instead of `readNBytes` we could read all bytes as before and truncate the byte array here?

Reply:
`readExclusive` means that this is the last operation on given stream. Why should we read all bytes to satisfy some contract if we only need N bytes? The stream being in "unpredictable" state is not our concern, because it will only ever be touched by the garbage collector. Now that we've arrived at this solution, I prefer fuzzing efficiency+updating the contract (readExclusive is the last operation on the stream). WDYT?